python/llm/src/ipex_llm/transformers/npu_pipeline_model/convert_pipeline.py

```diff
@@ -63,85 +63,121 @@ def generate(
+            else:
+                new_size = (
+                    val_.size(0),
+                    val_.size(1),
+                    val_.size(3),
+                    self.kv_len,
+                )
+                val_cache = val_.transpose(-1, -2)
+                val = val_cache.as_strided(new_size, val_cache.stride(), storage_offset=0)
+            key.to(torch.float16).numpy().tofile(os.path.join(temp_dir, f"key_cache_{layer}.bin"))
+            val.to(torch.float16).numpy().tofile(os.path.join(temp_dir, f"value_cache_{layer}.bin"))
+
+        token = input_id.to(torch.int32).item()
+        output_tokens.append(torch.tensor([token]))
+        if streamer is not None:
+            streamer.put(torch.tensor([token]))
 
-    if "eos_token_id" not in new_generate_kwargs:
-        eos = 0xffffffff
-    else:
-        eos = new_generate_kwargs["eos_token_id"]
+        if "eos_token_id" not in new_generate_kwargs:
+            eos = 0xffffffff
+        else:
+            eos = new_generate_kwargs["eos_token_id"]
 
-    bdata = bdata + eos.to_bytes(4, sys.byteorder)
+        time_t1 = time.perf_counter()
+        idx += 1
 
-    time_start = time.perf_counter()
+        # start generate_serve by Thread
+        thread = threading.Thread(target=generate_serve,
+                                  args=(self.kv_len, self.num_head,
+                                        self.head_dim, self.num_layers,
+                                        self.transpose_value_cache,
+                                        new_tokens - 2))
+        thread.start()
 
-    input_pipe.write(bytearray(bdata))
-    input_pipe.flush()
+        in_pipe_path = "\\\\.\\pipe\\llminputpipe"
+        out_pipe_path = "\\\\.\\pipe\\llmoutputpipe"
 
-    buffersize = 4
-    output_tokens = []
-    while True:
-        data = output_pipe.read(buffersize)
-        if len(data) == 0:
-            break
-        token = int.from_bytes(data, sys.byteorder)
-        idx += 1
-        if time_t1 is None:
-            time_t1 = time.perf_counter()
-        output_tokens.append(torch.tensor([token]))
-        if streamer is not None:
-            streamer.put(torch.tensor([token]))
-        if token == eos:
-            break
+        while True:
+            try:
+                input_pipe = open(in_pipe_path, "wb")
+            except:
+                print('Waiting for input pipe')
+                time.sleep(1)
+            else:
+                break
 
-    output = torch.stack(output_tokens, dim=1)
-    output = torch.cat((inputs, output), dim=1)
-    if streamer is not None:
-        streamer.end()
+        while True:
+            try:
+                output_pipe = open(out_pipe_path, "rb")
+            except:
+                print('Waiting for output pipe')
+                time.sleep(1)
+            else:
+                break
+
+        time_start = time.perf_counter()
+
+        bdata = str.encode(str(temp_dir))
```

Comment:
check length of `str(temp_dir)` here (<2000?)

Reply:
Added check here.